File: src/org/scripts/kotlin/content/commands/Jail.java
package org.scripts.kotlin.content.commands;

import com.runelive.model.PlayerRights;
import com.runelive.model.player.command.Command;
import com.runelive.util.Misc;
import com.runelive.world.World;
import com.runelive.world.entity.impl.player.Player;
import com.runelive.world.entity.impl.player.PlayerSaving;

/**
 * "The digital revolution is far more significant than the invention of writing or even of printing." - Douglas
 * Engelbart
 * Created on 7/28/2016.
 *
 * @author Seba
 */
public class Jail extends Command {

    public Jail(PlayerRights playerRights) {
        super(playerRights);
    }

    @Override
    public void execute(Player player, String[] args, PlayerRights privilege) {
        if (args.length != 1) {
            player.getPacketSender().sendMessage("Please use the command as ::jail-playername");
            return;
        }
        Player punishee = World.getPlayerByName(args[0]);
        /*PlayerSaving.accountExists(args[0], rs -> {
            if (rs.next()) {
                // account exists

                int cellAmounts = Misc.getRandom(1);
                switch (cellAmounts) {

                }
                when (cellAmounts) {
                    1 -> {
                        punishee.isJailed = true
                        punishee.moveTo(Position(1969, 5011, 0))
                    }
                    2 -> {
                        punishee.isJailed = true
                        punishee.moveTo(Position(1969, 5008, 0))
                    }
                    3 -> {
                        punishee.isJailed = true
                        punishee.moveTo(Position(1969, 5005, 0))
                    }
                    4 -> {
                        punishee.isJailed = true
                        punishee.moveTo(Position(1969, 5002, 0))
                    }
                    5 -> {
                        punishee.isJailed = true
                        punishee.moveTo(Position(1969, 4999, 0))
                    }
                    6 -> {
                        punishee.isJailed = true
                        punishee.moveTo(Position(1980, 5011, 0))
                    }
                    7 -> {
                        punishee.isJailed = true
                        punishee.moveTo(Position(1980, 5008, 0))
                    }
                    8 -> {
                        punishee.isJailed = true
                        punishee.moveTo(Position(1980, 5005, 0))
                    }
                    9 -> {
                        punishee.isJailed = true
                        punishee.moveTo(Position(1980, 5002, 0))
                    }
                    10 -> {
                        punishee.isJailed = true
                        punishee.moveTo(Position(1980, 4999, 0))
                    }
                }
            } else {
                player.packetSender.sendMessage("Player $jail_punishee does not exist.")
            }
        });*/

    }
}

File: src/org/scripts/kotlin/content/commands/Jail.kt
package org.scripts.kotlin.content.commands

import com.runelive.model.StaffRights
import com.runelive.model.Position
import com.runelive.model.player.command.Command
import com.runelive.util.Misc
import com.runelive.world.World
import com.runelive.world.entity.impl.player.Player

/**
 * "The digital revolution is far more significant than the invention of writing or even of printing." - Douglas
 * Engelbart
 * Created on 7/28/2016.

 * @author Seba
 */
class Jail(staffRights: StaffRights) : Command(staffRights) {

    override fun execute(player: Player, args: Array<String>?, privilege: StaffRights) {
        if (args == null) {
            player.packetSender.sendMessage("Example usage: ::jail-playername-time-timeunit")
            player.packetSender.sendMessage("Time units are M for minutes, H for hours, D for days.")
        } else if (args.size == 3) {
            val playerName = args[0]

            val victim = World.getPlayerByName(playerName)

            if (victim == null) {
                player.packetSender.sendMessage("The offending player is not currently online.")
                return
            }

            var time = -1
            try {
                time = Integer.parseInt(args[1])
            } catch (e: NumberFormatException) {
                player.packetSender.sendMessage("There was an error parsing your time arguement, try using numbers :).")
                return
            }

            if (time == -1 || time == 0) {
                return
            } else if (time > 60) {
                player.packetSender.sendMessage("Time amount cannot be more than 60.")
                return
            }

            val u = args[2].toLowerCase()

            val unit = u[0]

            if (unit == 'd') {
                if (player.staffRights != StaffRights.OWNER) {
                    player.packetSender.sendMessage("You must be an an owner to use the day parameter.")
                    return
                }
            }

            var calculatedTime = 0

            when (unit) {
                'm' -> calculatedTime = time * minute
                'h' -> calculatedTime = time * hour
                'd' -> calculatedTime = time * day
                else -> {
                    player.packetSender.sendMessage("Time units are M for minutes, H for hours, D for days. Please try again.")
                    return
                }
            }
            player.packetSender.sendMessage("You have sent " + victim.username + " to jail for " + time + " " + unit)
            val cellCords = coordinates[Misc.random(coordinates.size - 1)]
            val position = Position(cellCords[0], cellCords[1], 0)
            victim.moveTo(position)
            victim.playerTimers.jailTicks = calculatedTime
            victim.packetSender.sendMessage("You have been sent to jail for your bad deeds!")
        } else {
            player.packetSender.sendMessage("Example usage: ::jail-playername-time-timeunit")
            player.packetSender.sendMessage("Time units are M for minutes, H for hours, D for days.")
        }
    }

    companion object {

        val coordinates = arrayOf(intArrayOf(1969, 5011), intArrayOf(1969, 5008), intArrayOf(1969, 5005), intArrayOf(1969, 5002), intArrayOf(1969, 4999), intArrayOf(1980, 5011), intArrayOf(1980, 5008), intArrayOf(1980, 5005), intArrayOf(1980, 5002), intArrayOf(1980, 4999))

        val minute = Misc.secondsToTicks(60)

        val hour = minute * 60

        val day = hour * 24
    }
}
